### PR TITLE
chore: debugger improvements 2

### PR DESCRIPTION
**Pull request by DaniPopes**
## Motivation

Follow-up to #6491

No functional changes, just clean up around terminal init by using RAII.
Bad diff because of moving to new file :(

### CODE REVIEW

File: crates/anvil/src/service.rs [hunk 1]
```diff
@@ -106,7 +106,7 @@ type BlockMiningFuture =
     Pin<Box<dyn Future<Output = (MinedBlockOutcome, Arc<Backend>)> + Send + Sync>>;
 
 /// A type that exclusively mines one block at a time
-#[must_use = "BlockProducer does nothing unless polled"]
+#[must_use = "streams does nothing unless polled"]
```
Comment (mattsse) [inline crates/anvil/src/service.rs:109]:
```suggestion
#[must_use = "stream does nothing unless polled"]
```

File: crates/cli/src/utils/cmd.rs [hunk 2]
```diff
@@ -404,13 +404,12 @@ pub async fn handle_traces(
...
-            sources,
-            breakpoints: Default::default(),
-        };
-        debugger.run()?;
+        let mut debugger = DebuggerBuilder::new()
+            .debug_arena(&result.debug)
+            .decoder(&decoder)
+            .sources(sources)
+            .build()?;
+        debugger.try_run()?;
```
Comment (mattsse) [inline crates/cli/src/utils/cmd.rs:412]: I like this

File: crates/debugger/src/builder.rs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    /// Map of the debugger breakpoints.
+    breakpoints: Breakpoints,
+}
+
+impl DebuggerBuilder {
+    /// Creates a new debugger builder.
+    #[inline]
+    pub fn new() -> Self {
+        Self::default()
+    }
```
Comment (mattsse) [inline crates/debugger/src/builder.rs:31]: personally, I think new -> default is redundant

File: crates/evm/core/src/fork/backend.rs [hunk 4]
```diff
@@ -84,7 +84,7 @@ enum BackendRequest {
 ///
 /// This handler will remain active as long as it is reachable (request channel still open) and
 /// requests are in progress.
-#[must_use = "BackendHandler does nothing unless polled."]
+#[must_use = "futures does nothing unless polled"]
```
Comment (mattsse) [inline crates/evm/core/src/fork/backend.rs:87]:
```suggestion
#[must_use = "future does nothing unless polled"]
```

File: crates/evm/core/src/fork/multi.rs [hunk 5]
```diff
@@ -179,7 +179,7 @@ enum ForkTask {
 }
 
 /// The type that manages connections in the background
-#[must_use = "MultiForkHandler does nothing unless polled."]
+#[must_use = "futures does nothing unless polled"]
```
Comment (mattsse) [inline crates/evm/core/src/fork/multi.rs:182]:
```suggestion
#[must_use = "future does nothing unless polled"]
```

File: crates/anvil/src/service.rs (hunk 1 above)
Comment (mattsse) [reply crates/anvil/src/service.rs:109]: either this or `do nothing`

File: crates/evm/core/src/fork/backend.rs (hunk 4 above)
Comment (mattsse) [reply crates/evm/core/src/fork/backend.rs:87]: either this or `do nothing`

**Review by mattsse — APPROVED** [review]
this is a lot cleaner, ty.

only nits

File: crates/anvil/src/service.rs (hunk 1 above)
Comment (DaniPopes) [reply crates/anvil/src/service.rs:109]: All other `must_use` use plural

**Pull request merged**